tests/mongo-tools/helpers.go

```diff
@@ -82,3 +90,79 @@ func getDatabaseState(t *testing.T, ctx context.Context, db *mongo.Database) map
 
 	return dbState
 }
+
+// compareFiles takes two files and checks if they have a same content.
+// If they don't, it prints a short diff view.
+func compareFiles(t *testing.T, file1, file2 *os.File) {
+	t.Helper()
+	h := sha256.New()
+
+	_, err := io.Copy(h, file1)
+	require.NoError(t, err)
+
+	hash1 := h.Sum(nil)
+
+	h.Reset()
+
+	_, err = io.Copy(h, file2)
+	require.NoError(t, err)
+
+	hash2 := h.Sum(nil)
+
+	// compare hashes of both files
+	if assert.Equal(t, hash1, hash2, "Checksums of following files are different:", file1.Name(), file2.Name()) {
+		return
+	}
+
+	content1, err := io.ReadAll(file1)
+	require.NoError(t, err)
+
+	content2, err := io.ReadAll(file2)
+	require.NoError(t, err)
+
+	difflib.NewMatcher(
+		difflib.SplitLines(string(content1)),
+		difflib.SplitLines(string(content2)),
+	)
+}
+
+// compareDirs compares two directories and their files recursively.
+// It ignores paths based on regex expressions under ignorePathRegs.
+func compareDirs(t *testing.T, dir1, dir2 string, ignorePathRegs ...string) {
+	var regexps []*regexp.Regexp
+	for _, reg := range ignorePathRegs {
+		regexps = append(regexps, regexp.MustCompile(reg))
+	}
+
+	err := filepath.WalkDir(dir1, func(path string, d fs.DirEntry, err error) error {
+		assert.NoError(t, err)
+		comparePath := strings.Replace(path, dir1, dir2, 1)
+
+		// skip all ignored paths
+		for _, reg := range regexps {
+			if reg.MatchString(path) {
+				return nil
+			}
+		}
+
+		if d.IsDir() {
+			_, err = os.Stat(comparePath)
+			assert.NoError(t, err)
+			return nil
+		}
```

Comment:
It's because we don't want to treat directories as files and compare their nonexisting "content". The `WalkDir` function should check for the files of this directory at some iteration.

Reply:
Then I guess that should be just
```go
if d.IsDir() {
 return nil
}
```

I don't understand how `WalkDir` could find non-existing directory

Reply:
I think I don't understand. Let's say we have 2 directories that we want to compare: A and B.
```
├── A/
│   ├── boo/
│   └── foo/
│       └── bar.txt
└── B/
    ├── boo/
    └── foo/
        └── bar.txt
```
Walkdir that is running on A/ even when A/boo/ is empty will go through it, so it's worth to check if the B/boo/ also exists.

Reply:
Just so I understand, this is about testing _expected_ (_dir1_) having an empty directory but _actual_ (_dir2_) is not existing? I feel this isn't necessary since there is no files inside neither directory, we don't worry about if it exist or not.

What's more interesting is if _expected_ (_dir1_) have an empty directory but _actual_ (_dir2_) has files. But let's not go there in this task for now.